client/my-sites/themes/switching-homepage-modal.jsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Dialog } from '@automattic/components';
+import {
+	getCanonicalTheme,
+	hasActivatedTheme,
+	hasAutoLoadingHomepageFeature,
+	isActivatingTheme,
+	isThemeActive,
+	shouldShowHomepageWarning,
+	getPreInstallingThemeId,
+} from 'state/themes/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	acceptSwitchingHomepageWarning,
+	hideSwitchingHomepageWarning,
+	activate as activateTheme,
+} from 'state/themes/actions';
+
+/**
+ * Style dependencies
+ */
+import './switching-homepage-modal.scss';
+
+class SwitchingHomepageModal extends Component {
+	static propTypes = {
+		source: PropTypes.oneOf( [ 'details', 'list', 'upload' ] ).isRequired,
+		theme: PropTypes.shape( {
+			author: PropTypes.string,
+			author_uri: PropTypes.string,
+			id: PropTypes.string,
+			name: PropTypes.string,
+		} ),
+		hasActivated: PropTypes.bool.isRequired,
+		isActivating: PropTypes.bool.isRequired,
+		hasAutoLoadingHomepage: PropTypes.bool,
+		siteId: PropTypes.number,
+		isVisible: PropTypes.bool,
+		onClose: PropTypes.func,
+		installingThemeId: PropTypes.string.isRequired,
+	};
+
+	onCloseModal = ( activate = false ) => {
+		if ( activate ) {
+			const { installingThemeId, siteId, source } = this.props;
+			this.props.acceptSwitchingHomepageWarning( installingThemeId );
+			return this.props.activateTheme( installingThemeId, siteId, source );
+		}
+		this.props.hideSwitchingHomepageWarning();
+	};
+
+	onClickButtonHandler = activate => () => this.onCloseModal( activate );
```

Comment:
Why can't `onCloseModal()` not be called directly?

Reply:
Because there are two ways to close the modal. Accepting switching the homepage, or not.
It's defined by the `activated` parameter. You'd like to see how it works some lines below:

```es6
onClick: this.onClickButtonHandler( false ),
```

or

```es6
onClick: this.onClickButtonHandler( true ),
```

Reply:
And this could not be `onClick: this.onCloseModal( false )`?

Reply:
I guess so, :-D

Reply:
good point 👍